conf4j-spring/src/main/java/com/sabre/oss/conf4j/spring/handler/ConverterDecoratorBeanDefinitionParser.java

```diff
@@ -0,0 +1,85 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sabre.oss.conf4j.spring.handler;
+
+import org.springframework.beans.factory.support.AbstractBeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.xml.ParserContext;
+import org.w3c.dom.Element;
+
+import static com.sabre.oss.conf4j.spring.handler.AttributeConstants.CLASS_ATTRIBUTE;
+import static com.sabre.oss.conf4j.spring.handler.AttributeConstants.FACTORY_ATTRIBUTE;
+import static com.sabre.oss.conf4j.spring.handler.AttributeConstants.ORDER_ATTRIBUTE;
+import static java.lang.Integer.valueOf;
+import static org.springframework.util.StringUtils.hasText;
+
+public class ConverterDecoratorBeanDefinitionParser extends AbstractClassBeanDefinitionParser {
+    private static final String DEFAULT_DECORATING_CONVERTER_FACTORY = "com.sabre.oss.conf4j.spring.converter.DefaultDecoratingConverterFactory";
+
+    private final OrderingProxy proxy = new OrderingProxy();
+
+    @Override
+    protected BeanDefinitionBuilder getBeanDefinitionBuilder(Element element, ParserContext parserContext) {
+        String beanClassName = element.getAttribute(FACTORY_ATTRIBUTE);
+        boolean createFactory = false;
+        if (!hasText(beanClassName)) {
+            beanClassName = element.getAttribute(CLASS_ATTRIBUTE);
+            if (!hasText(beanClassName)) {
+                parserContext.getReaderContext().error("Attribute 'class' and attribute 'factory' must not be empty", element);
+                return null;
+            }
+            beanClassName = DEFAULT_DECORATING_CONVERTER_FACTORY;
+            createFactory = true;
+        }
+
+        String order = element.getAttribute(ORDER_ATTRIBUTE);
+        if (hasText(order)) {
+            beanClassName = proxy.create(beanClassName, valueOf(order));
+        }
+
+        BeanDefinitionBuilder builder = BeanDefinitionBuilder.genericBeanDefinition(beanClassName);
+        builder.getRawBeanDefinition().setSource(parserContext.extractSource(element));
+
+        if (createFactory) {
+            builder.addPropertyValue("converterClass", element.getAttribute(CLASS_ATTRIBUTE));
+        }
+
+        return builder;
+    }
+
+    @Override
+    protected String resolveId(Element element, AbstractBeanDefinition definition, ParserContext parserContext) {
+        String classId = super.resolveId(element, definition, parserContext);
+        if (!classId.equals(element.getAttribute(CLASS_ATTRIBUTE))) {
+            return classId;
+        }
+        String factoryId = element.getAttribute(FACTORY_ATTRIBUTE);
+        if (!hasText(factoryId)) {
+            // TODO Discuss naming convention for converter decorators created using DefaultDecoratingConverterFactory
+            return classId + "Factory";
```

Comment:
We must be sure that this name is unique. Maybe using `$` sign and longer suffix e.d. `$DecoratingFactory` or `$Conf4jConverterDecoratingFactory`? Is this name used for anything? If not we may let spring generate it automatically.